src/main/java/hudson/plugins/gradle/injection/MavenExtensionsHandler.java

```diff
@@ -8,78 +8,109 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
 import static hudson.plugins.gradle.injection.CopyUtil.copyResourceToNode;
-import static hudson.plugins.gradle.injection.MavenExtensionsHandler.MavenExtension.CCUD;
-import static hudson.plugins.gradle.injection.MavenExtensionsHandler.MavenExtension.GRADLE_ENTERPRISE;
+import static hudson.plugins.gradle.injection.CopyUtil.unsafeResourceDigest;
 
 public class MavenExtensionsHandler {
 
     static final String LIB_DIR_PATH = "jenkins-gradle-plugin/lib";
 
-    private final MavenExtensionFileHandler geExtensionHandler = new MavenExtensionFileHandler(GRADLE_ENTERPRISE);
-    private final MavenExtensionFileHandler ccudExtensionHandler = new MavenExtensionFileHandler(CCUD);
+    private final Map<MavenExtension, MavenExtensionFileHandler> fileHandlers =
+        Arrays.stream(MavenExtension.values())
+            .map(MavenExtensionFileHandler::new)
+            .collect(Collectors.toMap(h -> h.extension, Function.identity()));
```

Comment:
I think you can use Guava's `Maps.uniqueIndex` here to transform the MavenExtensionFileHandler stream in a map whose index is the `handler.extension`.

Reply:
Yep, it's also possible:

```
Map<MavenExtension, MavenExtensionFileHandler> fileHandlers =
    Maps.uniqueIndex(
        Iterables.transform(Arrays.asList(MavenExtension.values()), MavenExtensionFileHandler::new),
        h -> h.extension
    );
```

ATM there is a very old guava version (11.0), transitively coming from Jenkins, so we need to use relatively verbose `Iterables` syntax to do the transformation.    

Or we can use `Stream`

```
Map<MavenExtension, MavenExtensionFileHandler> fileHandlers =
    Maps.uniqueIndex(
        Arrays.stream(MavenExtension.values()).map(MavenExtensionFileHandler::new).iterator(),
        h -> h.extension
    );
```

In the end, it doesn't seem to be more readable/simpler to follow, due to the need for soma additional method calls. WDYT?

Reply:
ok I just saw it was already using guava, but if it's an old one, and coming via a transitive dependency, I'm fine keeping your Java8 approach.